content/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-php-project-for-codespaces.md

```diff
@@ -0,0 +1,209 @@
+  "image": "mcr.microsoft.com/devcontainers/php:0-8.2",
+
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  // "features": {},
+
+  // Configure tool-specific properties.
+  "customizations": {
+    // Configure properties specific to VS Code.
+    "vscode": {
+      "settings": {},
+      "extensions": [
+        "streetsidesoftware.code-spell-checker"
+      ]
+    }
+  },
+
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  // "forwardPorts": [8000],
+
+  // Use 'portsAttributes' to set default properties for specific forwarded ports. More info: https://code.visualstudio.com/docs/remote/devcontainerjson-reference.
+  "portsAttributes": {
+    "8000": {
+      "label": "Hello Remote World",
+      "onAutoForward": "notify"
+    }
+  }
+
+  // Use 'postCreateCommand' to run commands after the container is created.
+  // "postCreateCommand": "sudo chmod a+x \"$(pwd)\" && sudo rm -rf /var/www/html && sudo ln -s \"$(pwd)\" /var/www/html",
+
+  // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
+  // "remoteUser": "root"
+}
+```
+
+{% data reusables.codespaces.devcontainer-properties-1 %}
+{% data reusables.codespaces.devcontainer-properties-2 %}
+{% data reusables.codespaces.additional-container-config %}
+
+## Step 3: Modify your devcontainer.json file
+
+With your dev container configuration added and a basic understanding of what everything does, you can now make changes to customize your environment further. In this example, you'll add properties that will:
+* Run `composer install`, after the dev container is created, to install the dependencies listed in the `composer.json` file.
+* Automatically install a {% data variables.product.prodname_vscode_shortname %} extension in this codespace.
+
+1. In the `devcontainer.json` file, add a comma after the `portsAttributes` property.
+
+   ```json{:copy}
+   // Use 'portsAttributes' to set default properties for specific forwarded ports. More info: https://code.visualstudio.com/docs/remote/devcontainerjson-reference.
+     "portsAttributes": {
+       "8000": {
+         "label": "Hello Remote World",
+         "onAutoForward": "notify"
+       }
+   },
+   ```
+
+1. Uncomment the `postCreateCommand` property and assign it the command `composer install`.
+
+   ```json{:copy}
+   // Use 'postCreateCommand' to run commands after the container is created.
+   "postCreateCommand": "composer install",
+   ```
+
+1. Edit the `customizations` property as follows to install the "Composer" extension.
+
+   ```json{:copy}
+   // Configure tool-specific properties.
+   "customizations": {
+     // Configure properties specific to VS Code.
+     "vscode": {
+       "settings": {},
+       "extensions": [
+         "streetsidesoftware.code-spell-checker",
+         "ikappas.composer"
+       ]
+     }
+   },
+   ```
```

Comment:
I think this should follow the file in top-down order of things that need changed:

1. In the `devcontainer.json` file, delete the two commented-out lines about features: 

   ```json
   // Features to add to the dev container. More info: https://containers.dev/features.
   // "features": {},
   ```

1. Edit the `customizations` property as follows to install the "Composer" extension.

   ```json{:copy}
   // Configure tool-specific properties.
   "customizations": {
     // Configure properties specific to VS Code.
     "vscode": {
       "extensions": [
         "ikappas.composer"
       ]
     }
   },
   ```

1. Add a comma after the `features` property.

   ```json{:copy}
   "features": {
     "ghcr.io/devcontainers/features/github-cli:1": {}
   },
   ```

1. Uncomment the `postCreateCommand` property and change its value to the command `composer install`.

   ```json{:copy}
   // Use 'postCreateCommand' to run commands after the container is created.
   "postCreateCommand": "composer install"
   ```